File: source/clibCore/Models/Context/MiniCollectionModel.vb

Option Explicit On
Option Strict On

Imports System
Imports System.Collections.Generic
Imports System.Text
Imports Contensive.BaseClasses
Imports Contensive.Core.Controllers
Imports Newtonsoft.Json

Namespace Contensive.Core.Models.Entity
    '
    '
    '====================================================================================================
    ''' <summary>
    ''' miniCollection - This is an old collection object used in part to load the cdef part xml files. REFACTOR this into CollectionWantList and werialization into jscon
    ''' </summary>
    Public Class miniCollectionModel
        Implements ICloneable
        '
        ' Content Definitions (some data in CDef, some in the CDef extension)
        '
        Public name As String
        Public isBaseCollection As Boolean                     ' true only for the one collection created from the base file. This property does not transfer during addSrcToDst
        Public CDef As New Dictionary(Of String, Models.Complex.cdefModel)
        Public SQLIndexCnt As Integer
        Public SQLIndexes() As SQLIndexType
        Public Structure SQLIndexType
            Dim DataSourceName As String
            Dim TableName As String
            Dim IndexName As String
            Dim FieldNameList As String
            Dim dataChanged As Boolean
        End Structure
        Public MenuCnt As Integer
        Public Menus() As MenusType
        Public Structure MenusType
            Dim Name As String
            Dim IsNavigator As Boolean
            Dim menuNameSpace As String
            Dim ParentName As String
            Dim ContentName As String
            Dim LinkPage As String
            Dim SortOrder As String
            Dim AdminOnly As Boolean
            Dim DeveloperOnly As Boolean
            Dim NewWindow As Boolean
            Dim Active As Boolean
            Dim AddonName As String
            Dim dataChanged As Boolean
            Dim Guid As String
            Dim NavIconType As String
            Dim NavIconTitle As String
            Dim Key As String
        End Structure
        Public AddOns() As AddOnType
        Public Structure AddOnType
            Dim Name As String
            Dim Link As String
            Dim ObjectProgramID As String
            Dim ArgumentList As String
            Dim SortOrder As String
            Dim Copy As String
            Dim dataChanged As Boolean
        End Structure
        Public AddOnCnt As Integer
        Public Styles() As StyleType
        Public Structure StyleType
            Dim Name As String
            Dim Overwrite As Boolean
            Dim Copy As String
            Dim dataChanged As Boolean
        End Structure
        Public StyleCnt As Integer
        Public StyleSheet As String
        Public ImportCnt As Integer
        Public collectionImports() As ImportCollectionType
        Public Structure ImportCollectionType
            Dim Name As String
            Dim Guid As String
        End Structure
        Public PageTemplateCnt As Integer
        Public PageTemplates() As PageTemplateType
        '   Page Template - started, but CDef2 and LoadDataToCDef are all that is done do far
        Public Structure PageTemplateType
            Dim Name As String
            Dim Copy As String
            Dim Guid As String
            Dim Style As String
        End Structure
        Public Function Clone() As Object Implements ICloneable.Clone
            Return Me.MemberwiseClone
        End Function
    End Class
End Namespace

File: source/clibCoreCs/Models/Context/miniCollectionModel.cs

using System;
using System.Reflection;
using System.Xml;
using System.Diagnostics;
using System.Linq;
using System.Text.RegularExpressions;
using System.Collections.Generic;
using System.Data;
using System.Data.SqlClient;
using Contensive.Core;
using Contensive.Core.Models.Entity;
using Contensive.Core.Controllers;
using static Contensive.Core.Controllers.genericController;
using static Contensive.Core.constants;
//
namespace Contensive.Core.Models.Entity {
    //
    //
    //====================================================================================================
    /// <summary>
    /// miniCollection - This is an old collection object used in part to load the cdef part xml files. REFACTOR this into CollectionWantList and werialization into jscon
    /// </summary>
    public class miniCollectionModel : ICloneable {
        //
        // Content Definitions (some data in CDef, some in the CDef extension)
        //
        public string name;
        public bool isBaseCollection; // true only for the one collection created from the base file. This property does not transfer during addSrcToDst
        public Dictionary<string, Models.Complex.cdefModel> CDef = new Dictionary<string, Models.Complex.cdefModel>();
        //public int SQLIndexCnt;
        public List<SQLIndexType> SQLIndexes;
        public struct SQLIndexType {
            public string DataSourceName;
            public string TableName;
            public string IndexName;
            public string FieldNameList;
            public bool dataChanged;
        }
        public int MenuCnt;
        public MenusType[] Menus;
        public struct MenusType {
            public string Name;
            public bool IsNavigator;
            public string menuNameSpace;
            public string ParentName;
            public string ContentName;
            public string LinkPage;
            public string SortOrder;
            public bool AdminOnly;
            public bool DeveloperOnly;
            public bool NewWindow;
            public bool Active;
            public string AddonName;
            public bool dataChanged;
            public string Guid;
            public string NavIconType;
            public string NavIconTitle;
            public string Key;
        }
        public AddOnType[] AddOns;
        public struct AddOnType {
            public string Name;
            public string Link;
            public string ObjectProgramID;
            public string ArgumentList;
            public string SortOrder;
            public string Copy;
            public bool dataChanged;
        }
        public int AddOnCnt;
        public StyleType[] Styles;
        public struct StyleType {
            public string Name;
            public bool Overwrite;
            public string Copy;
            public bool dataChanged;
        }
        public int StyleCnt;
        public string StyleSheet;
        public int ImportCnt;
        public ImportCollectionType[] collectionImports;
        public struct ImportCollectionType {
            public string Name;
            public string Guid;
        }
        public int PageTemplateCnt;
        public PageTemplateType[] PageTemplates;
        //   Page Template - started, but CDef2 and LoadDataToCDef are all that is done do far
        public struct PageTemplateType {
            public string Name;
            public string Copy;
            public string Guid;
            public string Style;
        }
        public object Clone() {
            return this.MemberwiseClone();
        }
    }
}
